File: application-web-mvc/src/main/java/com/osori/mvc/domain/Account.java
package com.osori.mvc.domain;

import lombok.Data;
import org.springframework.security.crypto.password.PasswordEncoder;

import javax.persistence.*;

@Entity
@Data
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(unique = true)
    private String username;
    private String password;
    private String role;


    public void encodePassword(PasswordEncoder passwordEncoder) {
        this.password = passwordEncoder.encode(this.password);
    }
}

File: application-web-mvc/src/main/kotlin/com/osori/mvc/domain/Account.kt
package com.osori.mvc.domain

import jakarta.persistence.*
import org.springframework.security.crypto.password.PasswordEncoder

@Entity
data class Account(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null,

    @Column(unique = true)
    var username: String? = null,
    var password: String? = null,
    var role: String? = null
) {
    fun encodePassword(passwordEncoder: PasswordEncoder) {
        password = passwordEncoder.encode(password)
    }
}